lib/Dialect/TfheRustBool/IR/TfheRustBoolTypes.td

```diff
@@ -28,4 +28,27 @@ def TfheRustBool_ServerKey : TfheRustBool_Type<"ServerKey", "server_key", [PassB
   let summary = "The boolean server key required to perform homomorphic operations.";
 }
 
+
+class TfheRustBool_BooleanGate<string gate>
+  : TfheRustBool_Type<"BooleanGate" # gate, "bin_gate" # gate> {
+  let summary = "Enum placeholder to define the binary gate" # gate # " to perform as homomoprhic operation.";
+}
+
+foreach i = ["And", "Or", "Xor", "Nand", "Nor", "Xnor"] in {
```

Comment:
this is awesome! (any reason not do something similar for the CGGI attributes? is it more difficult to do with a string attr enum?)